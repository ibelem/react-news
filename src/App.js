import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';


const list = [
  { title: 'React',
    url: 'https://facebook.github.io/react',
    author: 'Jordan Walke',
    num_comments: 3,
    points: 4,
    objectID: 0,
  },
  { title: 'Redux',
    url: 'https://github.com/reactjs/redux',
    author: 'Dan Abramov, Andrew Clark',
    num_comments: 2,
    points: 5,
    objectID: 100,
  },  
];

class App extends Component {

  constructor(pros) {
    super(pros);
  }

  render() {
    return (
      <div className='App'>
        { list.map(item =>  
          <div key={item.objectID}>
            <a href={item.url}>{item.objectID} {item.title}</a><br/>
            <span>{item.num_comments}</span><br/>
            <span>{item.points}</span>
          </div>
        )}
      </div>
    );
  }
}

export default App;
