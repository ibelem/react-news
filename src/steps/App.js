import React, { Component } from 'react';
// import logo from './logo.svg';
import './App.css';


const list = [
  {
    title: '中国',
    url: 'https://facebook.github.io/react',
    author: '民族',
    num_comments: 3,
    points: 4,
    objectID: 1,
  },
  {
    title: '世界',
    url: 'https://github.com/reactjs/redux',
    author: '全球',
    num_comments: 20,
    points: 5,
    objectID: 2,
  },
  {
    title: 'React',
    url: 'https://github.com/reactjs/redux',
    author: 'Daniel',
    num_comments: 2,
    points: 5,
    objectID: 3,
  },
  {
    title: 'Redux',
    url: 'https://github.com/reactjs/redux',
    author: 'Dan Abramov, Andrew Clark',
    num_comments: 2,
    points: 5,
    objectID: 4,
  },
];

const isSearched = searchTerm => item => item.title.toLowerCase().includes(searchTerm.toLowerCase());

class App extends Component {

  constructor(props) {
    super(props);

    this.state = {
      list,
      searchTerm: ''
    };

    this.onDismiss = this.onDismiss.bind(this);
    this.onClickMe1 = this.onClickMe1.bind(this); //FB preferred
    //this.onClickMe2 - No necessary to bind here
    this.onSearchChange = this.onSearchChange.bind(this);
  }

  onClickMe1() {
    console.log(this);
  }

  onClickMe2 = () => {
    console.log(this);
  }

  onDismiss(id) {
    const isNotId = item => item.objectID !== id;
    const updatedList = this.state.list.filter(isNotId);
    this.setState({ list: updatedList });
  }

  onSearchChange(event) {
    this.setState({ searchTerm: event.target.value })
  }

  // render() {
  //   return (
  //     <div className='App'>
  //       { this.state.list.map(item =>  
  //         <div key={item.objectID}>
  //           <span><a href={item.url}>{item.objectID} {item.title}</a></span> 
  //           <span>{item.author}</span> 
  //           <span>{item.num_comments}</span> 
  //           <span>{item.points}</span>
  //           <span>
  //             <button onClick={ () => this.onDismiss(item.objectID)} type='button'>
  //              Dismiss
  //             </button>
  //           </span>
  //         </div>
  //       )}
  //   );
  // }

  render() {
    const { searchTerm, list } = this.state;
    return (
      <div className='App'>
        {list.map(item => {
          const onHandlerDismiss = () => {
            this.onDismiss(item.objectID)
          }

          return (
            <div key={item.objectID}>
              <span><a href={item.url}>{item.objectID} {item.title}</a></span>
              <span>{item.author}</span>
              <span>{item.num_comments}</span>
              <span>{item.points}</span>
              <span>
                <button onClick={onHandlerDismiss} type='button'>
                  Dismiss
              </button>
              </span>
            </div>
          )

        }

        )}

        <button onClick={() => this.onClickMe1()} type='button'>
          Click Me 1
        </button>

        <button onClick={this.onClickMe2} type='button'>
          Click Me 2
        </button>

        <button onClick={console.log('CHINA3')} type='button'>
          Click Me 3
        </button>

        <button onClick={() => { console.log('CHINA4') }} type='button'>
          Click Me 4
        </button>

        <form>
          <input type='text' onChange={ this.onSearchChange } value={searchTerm} />
        </form>
        { list.filter(isSearched(this.state.searchTerm)).map(
          item => item.title
        )}
      </div>
    );
  }
}

export default App;
