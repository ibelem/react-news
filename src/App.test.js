import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App, { Button, Search, Table } from './App';

Enzyme.configure({ adapter: new Adapter() });

describe('App', ()=> {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', ()=> {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('Button', ()=> {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button>Test</Button>, div);
    // ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', ()=> {
    const component = renderer.create(<Button>Test</Button>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('Search', ()=> {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search>Search</Search>, div);
    // ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', ()=> {
    const component = renderer.create(<Search>Search</Search>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('Table', ()=> {

  const props = {
    list: [
      { title: '1', author: '1', num_comments: 1, points: 2, objectID: 'y' },
      { title: '2', author: '2', num_comments: 1, points: 2, objectID: 'z' },
      { title: '3', author: '3', num_comments: 2, points: 3, objectID: 'x' },
    ]
  }

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Table {...props} />, div);
    // ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', ()=> {
    const component = renderer.create(<Table {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows three items in list', ()=> {
    const element = shallow(
      <Table {...props} />
    );
    expect(element.find('.table-row').length).toBe(3);
  });

});

