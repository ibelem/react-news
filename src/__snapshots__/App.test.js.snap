// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="page"
>
  <div
    className="interactions"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value="react"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <div
    className="table"
  />
  <div
    className="interactions"
  >
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      More
    </button>
  </div>
</div>
`;

exports[`Button has a valid snapshot 1`] = `
<button
  className=""
  onClick={undefined}
  type="button"
>
  Test
</button>
`;

exports[`Search has a valid snapshot 1`] = `
<form
  onSubmit={undefined}
>
  <input
    onChange={undefined}
    type="text"
    value={undefined}
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;

exports[`Table has a valid snapshot 1`] = `
<div
  className="table"
>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "60%",
        }
      }
    >
      <a
        href={undefined}
      >
        y
         
        1
      </a>
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      1
    </span>
    <span
      style={
        Object {
          "width": "5%",
        }
      }
    >
      1
    </span>
    <span
      style={
        Object {
          "width": "5%",
        }
      }
    >
      2
    </span>
    <span>
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
    </span>
  </div>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "60%",
        }
      }
    >
      <a
        href={undefined}
      >
        z
         
        2
      </a>
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      2
    </span>
    <span
      style={
        Object {
          "width": "5%",
        }
      }
    >
      1
    </span>
    <span
      style={
        Object {
          "width": "5%",
        }
      }
    >
      2
    </span>
    <span>
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
    </span>
  </div>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "60%",
        }
      }
    >
      <a
        href={undefined}
      >
        x
         
        3
      </a>
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      3
    </span>
    <span
      style={
        Object {
          "width": "5%",
        }
      }
    >
      2
    </span>
    <span
      style={
        Object {
          "width": "5%",
        }
      }
    >
      3
    </span>
    <span>
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
    </span>
  </div>
</div>
`;
